docs/templates/tour.md

```diff
@@ -0,0 +1,324 @@
+  os                     = "linux"
+  startup_script_timeout = 180
+  startup_script         = <<-EOT
+    set -e
+
+    # install and start code-server
+    curl -fsSL https://code-server.dev/install.sh | sh -s -- --method=standalone --prefix=/tmp/code-server --version 4.11.0
+    /tmp/code-server/bin/code-server --auth none --port 13337 >/tmp/code-server.log 2>&1 &
+  EOT
+
+  metadata {
+    display_name = "CPU Usage"
+    key          = "0_cpu_usage"
+    script       = "coder stat cpu"
+    interval     = 10
+    timeout      = 1
+  }
+
+  metadata {
+    display_name = "RAM Usage"
+    key          = "1_ram_usage"
+    script       = "coder stat mem"
+    interval     = 10
+    timeout      = 1
+  }
+}
+```
+Agents can also run startup scripts, set environment variables and
+provide `metadata`. `
+
+These example `metadata` blocks are optional. Coder displays this
+information in the Coder dashboard. For basic information, you can
+invoke the [`coder stat`](../cli.md) command. If you need more
+control, you can write your own script.
+
+
+## 4. coder_app
+
+A
+['coder_app'](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/app)
+resource lets a developer use an app from the workspace's Coder
+dashboard.
+
+![Apps in a Coder workspace](../images/templates/workspace-apps.png)
+
+This is commonly used for [web IDEs](../ides/web-ides.md) such as
+[code-server](https://coder.com/docs/code-server/latest), RStudio, and
+JupyterLab.
+
+To install and run an app in the workspace, add it to the
+`startup_script` argument in `coder_agent` then add a `coder_app`
+resource to make it available in the workspace. See [web
+IDEs](../ides/web-ides.md) for some examples.
+
+```hcl
+resource "coder_app" "code-server" {
+  agent_id     = coder_agent.main.id
+  slug         = "code-server"
+  display_name = "code-server"
+  url          = "http://localhost:13337/?folder=/home/${local.username}"
+  icon         = "/icon/code.svg"
+  subdomain    = false
+  share        = "owner"
+
+  healthcheck {
+    url       = "http://localhost:13337/healthz"
+    interval  = 5
+    threshold = 6
+  }
+}
+```
+
+You can also use a `coder_app` resource to link to
+external apps, such as links to wikis or cloud consoles.
+
+```hcl
+resource "coder_app" "coder-server-doc" {
+  agent_id     = coder_agent.main.id
+  icon         = "/emojis/1f4dd.png"
+  slug         = "getting-started"
+  url          = "https://coder.com/docs/code-server"
+  external     = true
+}
+```
+
+## 5. Persistent storage
+
+We want our workspace's home directory to persist after the workspace
+is stopped so that a developer can continue their work when they start
+the workspace again.
+
+We do this in 2 parts:
+
+- Our `docker_volume` resource uses the `ignore_changes` argument
+- Later, our `docker_container` resource uses the Terraform [count](https://developer.hashicorp.com/terraform/language/meta-arguments/count)
+meta-argument.
```

Comment:
We use `count` only on ephemeral resources.

Reply:
Good catch. Thanks!